### PR TITLE
Updated to Swift 3 04-25

### LINKED ISSUE #13: Emits warnings on Swift 3 04-25
```
/Users/honzadvorsky/Documents/vapor/Packages/JSON-0.5.0/Source/JSONInterchangeDataParser.swift:140:55: warning: @autoclosure is now an attribute on a parameter type, instead of on the parameter itself
    private func parseSymbol(_ target: StaticString, @autoclosure _ iftrue: Void -> StructuredData) throws -> StructuredData {
                                                     ~^~~~~~~~~~~~
                                                                            @autoclosure
<unknown>:0: warning: @noescape is now an attribute on a parameter type, instead of on the parameter itself
/Users/honzadvorsky/Documents/vapor/Packages/JSON-0.5.0/Source/JSONInterchangeDataParser.swift:140:55: warning: @autoclosure is now an attribute on a parameter type, instead of on the parameter itself
    private func parseSymbol(_ target: StaticString, @autoclosure _ iftrue: Void -> StructuredData) throws -> StructuredData {
                                                     ~^~~~~~~~~~~~
                                                                            @autoclosure
<unknown>:0: warning: @noescape is now an attribute on a parameter type, instead of on the parameter itself
/Users/honzadvorsky/Documents/vapor/Packages/JSON-0.5.0/Source/JSONInterchangeDataParser.swift:140:55: warning: @autoclosure is now an attribute on a parameter type, instead of on the parameter itself
    private func parseSymbol(_ target: StaticString, @autoclosure _ iftrue: Void -> StructuredData) throws -> StructuredData {
                                                     ~^~~~~~~~~~~~
                                                                            @autoclosure
<unknown>:0: warning: @noescape is now an attribute on a parameter type, instead of on the parameter itself
/Users/honzadvorsky/Documents/vapor/Packages/JSON-0.5.0/Source/JSONParser.swift:134:55: warning: @autoclosure is now an attribute on a parameter type, instead of on the parameter itself
    private func parseSymbol(_ target: StaticString, @autoclosure _ iftrue: Void -> JSON) throws -> JSON {
                                                     ~^~~~~~~~~~~~
                                                                            @autoclosure
<unknown>:0: warning: @noescape is now an attribute on a parameter type, instead of on the parameter itself
/Users/honzadvorsky/Documents/vapor/Packages/JSON-0.5.0/Source/JSONParser.swift:134:55: warning: @autoclosure is now an attribute on a parameter type, instead of on the parameter itself
    private func parseSymbol(_ target: StaticString, @autoclosure _ iftrue: Void -> JSON) throws -> JSON {
                                                     ~^~~~~~~~~~~~
                                                                            @autoclosure
<unknown>:0: warning: @noescape is now an attribute on a parameter type, instead of on the parameter itself
/Users/honzadvorsky/Documents/vapor/Packages/JSON-0.5.0/Source/JSONParser.swift:134:55: warning: @autoclosure is now an attribute on a parameter type, instead of on the parameter itself
    private func parseSymbol(_ target: StaticString, @autoclosure _ iftrue: Void -> JSON) throws -> JSON {
                                                     ~^~~~~~~~~~~~
                                                                            @autoclosure
<unknown>:0: warning: @noescape is now an attribute on a parameter type, instead of on the parameter itself
```

Compilation still finishes, but it'd be nice to have a 04-25 clean version 😊

### PR DESCRIPTION
**Pull request by czechboy0**
Fixes #13. Please tag after merging 😊

### CODE REVIEW

File: Source/JSONParser.swift [hunk 1]
```diff
@@ -131,7 +131,7 @@ extension GenericJSONParser {
         return Character(UnicodeScalar(currentChar))
     }
 
-    private func parseSymbol(_ target: StaticString, @autoclosure _ iftrue: Void -> JSON) throws -> JSON {
+    private func parseSymbol(_ target: StaticString, _ iftrue: @autoclosure Void -> JSON) throws -> JSON {
```
Comment (czechboy0) [inline Source/JSONParser.swift:134]: The length of the line didn't change. Go home @houndci-bot, you're drunk.

**Comment by paulofaria** [comment]
you should update `.swift-version` file as well.

File: Source/JSONParser.swift (hunk 1 above)
Comment (tanner0101) [inline Source/JSONParser.swift:134]: Hound only analyzes diffs. Perhaps we should break this into multiple lines to appease it?

I'm a little ambivalent about the whole line length requirement tbh.

Comment (paulofaria) [inline Source/JSONParser.swift:134]: Looks like we can customize the style Hound enforces.

**Comment by czechboy0** [comment]
Sorry, done.

**Pull request merged**